pkg/ingester/ingester.go

```diff
@@ -178,6 +186,30 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	cfg.DefaultLimits.RegisterFlags(f)
 
 	f.StringVar(&cfg.IgnoreSeriesLimitForMetricNames, "ingester.ignore-series-limit-for-metric-names", "", "Comma-separated list of metric names, for which the -ingester.max-global-series-per-metric limit will be ignored. Does not affect the -ingester.max-global-series-per-user limit.")
+
+	f.BoolVar(&cfg.UtilizationBasedLimitingEnabled, "ingester.utilization-based-limiting-enabled", false, "Enable CPU/memory utilization based read path request limiting")
+	f.Float64Var(&cfg.CPUUtilizationTarget, "ingester.cpu-utilization-target", 0, "CPU target, as a fraction of 1, for CPU/memory utilization based request limiting")
```

Comment:
Naming-wise, I think "target" is not much clear. The reason is that what we set here in practice is not a target utilization (e.g. I want the ingester to consistently use X CPU cores) but we set a value which will be used to compute a limit along with `-ingester.read-path-utilization-target-ratio`.

I'm wondering if we're making config too complicated and we should just allow to directly set the limit instead, so "read-path-cpu-utilization-limit" and "read-path-memory-utilization-limit" (which would be my preference to begin with).

Probably this config was inherited from the initial version which had a write-path limit too, but since we're just going to have a read-path limit for now, I would simplify the config too. Being categorised as experimental, we can always change it with no notice if we later decide to add write-path limit too.